lib/Image.php

```diff
@@ -127,6 +127,36 @@ protected function get_dimensions_loaded( $dim ) {
 		return $this->width();
 	}
 
+	/**
+	 * Retrieve dimensions from SVG file
+	 *
+	 * @internal
+	 * @param string $svg SVG Path
+	 * @return array
+	 */
+	protected function get_dimensions_svg( $svg ) {
+		$svg    = simplexml_load_file( $svg );
+		$width  = '0';
+		$height = '0';
+
+		if ( false !== $svg ) {
+			$attributes = $svg->attributes();
+			if ( isset( $attributes->viewBox ) ) {
+				$viewbox = explode( ' ', $attributes->viewBox );
+				$width   = $viewbox[2];
+				$height  = $viewbox[3];
+			} elseif ( $attributes->width && $attributes->height ) {
+				$width  = (string) $attributes->width;
+				$height = (string) $attributes->height;
+			}
+		}
+
+		return (object) array(
+			'width'  => $width,
+			'height' => $height,
+		);
```

Comment:
I don't know how far down this rabbit hole we want to go, but parsing XML can be expensive and there's a further opportunity for more caching here in case someone calls `get_dimensions()` a bunch of times. The strategy is basically equivalent to how the transient API works:

```php
static $svg_dimensions;
$svg_dimensions = $svg_dimensions ?? [];

$hash = md5($svg);
if (isset($svg_dimensions[$hash])) {
  return $svg_dimensions[$hash];
}

$dimensions = /* ...normal computation... */;

$svg_dimensions[$hash] = $dimensions;

return $dimensions;
```

Reply:
This is def. the way to go for a grand plan (with the same strategy for caching regular image width/height which might be similarly expensive). I think we should proceed to catch up on functionality for now. But after, @vyskoczilova — would you be able to do an optimization PR using the guidance here from @acobster ?

Reply:
... for the next PR we should focus solely on 2.x

Reply:
@jarednova I would have a look at it (but in May, I'm totally overloaded). What's the easiest way to work with 2.0?

Reply:
no big diff there from 1.x! You can pull the branch here: https://github.com/timber/timber/tree/2.x and build a PR off of it. Tests and other processes behave the same. The biggest change is really the "regular user" install where the WP plugin is no longer available (requiring composer installation as a theme dependency). More info here:

https://timber.github.io/docs/v2/installation/